juspay/hyperswitch
# fix: added find all support for pm kv

## Type of Change


- [ x] Bugfix

## Description


Issue scenario faced:
1. create payment method (in /payments)
2. list payment method by customer (find all query that always went to db)
 above api returns empty list
 
 to handle cases where payment method entry present in kv but not drained to db yet, we have added a lookup in kv as well and combining data for findall from both kv and db

### Additional Changes





## Motivation and Context

list payment method by customer and merchant was eventually consistent when payment method followed kv path,
change here is to combine values from db and redis

## How did you test it?


1. create payment method and drain data to db (can skip this step if payment method entries are already present in db but not kv for the customer <> merchant)
2. create another payment method through kv
3. list payment methods should reflect pm created through kv as well
`curl --location 'http://localhost:8080/customers/<cust_id>/payment_methods' \
--header 'Accept: application/json' \
--header 'api-key:***'`


## Checklist
